File: projects/03/LetterGrade.java
/* LetterGrade.java is a driver for function letterGrade().
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed and Edited by by: Quentin Barnes
 * Date: Feb 23, 2019
 ****************************************************************/

import java.util.Scanner;

public class LetterGrade {

   public static void main(String[] args) {
      Scanner keyboard = new Scanner(System.in);
      System.out.println("\nEnter your grade average: "); // prompt

      int average = Integer.parseInt(keyboard.next());      // read year
      System.out.println(letterGrade(average)); // display its code
   }

   /***************************************************************
    * letterGrade() converts a grade average to its coresponding letter grade
    *
    * Receive: average, an int
    * PRE: average is <= 100 and >= 0
    * Return: A letter grade A, B, C, D, or F
    */

    public static char letterGrade(int average){
      switch (average / 10)
      {
         case 10: case 9:
            return 'A';
         case 8:
            return 'B';
         case 7:
            return 'C';
         case 6:
            return 'D';
         default:
            return 'F';
      }
   }

}

File: projects/03/clojure/src/letter_grade.clj
;;;; letter_grade.clj is a driver for function letterGrade.
;;;;
;;;; Input: average: an int that is <= 100 and >= 0
;;;; Output: A letter grade A, B, C, D, or F
;;;;
;;;; Usage: clojure -m letter_grade
;;;;
;;;; Begun by: Prof. Adams, for CS 214 at Calvin College.
;;;; Completed and Edited by by: Quentin Barnes
;;;; Date: Feb 23, 2019
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

(ns letter_grade)                    ; name the program

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; letterGrade() returns the code for a given grade average.
;;; Receive: average, an int.
;;; Precondition: average is <= 100 and >= 0
;;; Return: A letter grade A, B, C, D, or F
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;

;; solution using the cond function
(defn letterGrade [average]
  (cond
    (= average 10)   "A"
    (= average 9)    "A"
    (= average 8)    "B"
    (= average 7)    "C"
    (= average 6)    "D"
    :else            "F"
  )
)

;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
;;; Function -main() test-drives our letterGrade() function.
;;; Input: average: an int that is <= 100 and >= 0
;;; Output: A letter grade A, B, C, D, or F
;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
(defn -main []
 (print "\nEnter your grade average: ") (flush)
 (let
    [ average (read-line) ]                ; read year (a string)
    (println "Your average is: " (letterGrade ( quot (Integer/parseInt average) 10) ) )      ; display its code
 )
)

